File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 8/20/2026, 4:17:05 PM
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0 || (x%10 ==0 && x!=0)){
            return false;
        }
        
        int revertedNumber = 0;
        while(x > revertedNumber){
            revertedNumber = revertedNumber * 10 + x%10;
            x = x/10;
        }
        return x == revertedNumber || x == revertedNumber/10;
    }
}

File: 9-PalindromeNumber/9-PalindromeNumber.py
# Last updated: 8/20/2026, 4:17:10 PM
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x>=0:
            y=str(x)
            if y[::1]== y[::-1]:
                return True
        else:
            return False